dfsclient/src/main/java/org/hpccsystems/dfs/client/RowServiceInputStream.java

```diff
@@ -738,114 +737,87 @@ private int startFetch()
         }
 
         //------------------------------------------------------------------------------
-        // Read the response length from the request
+        // Read the response from the request
         //------------------------------------------------------------------------------
 
-        int len = 0;
-        try
-        {
-            len = readReplyLen();
-        }
-        catch (HpccFileException e)
+        boolean inTokenRetry = false;
+        do
         {
-            prefetchException = e;
+            RowServiceResponse response = null;
             try
             {
-                close();
+                response = readResponse();
             }
-            catch(Exception ie){}
-            return -1;
-        }
-
-        // If len == 0 we have finished reading the file
-        if (len == 0)
-        {
-            try
+            catch (HpccFileException e)
             {
-                close();
+                prefetchException = e;
```

Comment:
It will get handled, but since that code can be executed by the prefetch thread we postpone handling the exception until the reading thread can process it.